types/json-api-serializer/tslint.json

```diff
@@ -0,0 +1,6 @@
+{
+    "extends": "dtslint/dt.json",
+    "rules": {
+        "npm-naming": [true,{"mode":"code","errors":[["NeedsExportEquals",false]]}]
```

Comment:
[This](https://unpkg.com/browse/json-api-serializer@2.6.2/index.js) is not a default export. This lint rule is trying to tell you that your definition is incorrect. It needs to say `export = JSONAPISerailizer` (which means you will need to wrap all your exported types in a namespace also named `JSONAPISerializer`) in order to continue exporting them..

Reply:
Thanks for clarifying! I suspected as much but never got it to work any other way.

This is how you use the lib:

```typescript
import JSONAPISerializer from 'json-api-serializer';

const serializer = new JSONAPISerializer();
```

I tried to figure out how to add a namespace and still make that signature work. Sorry for bothering, but I'd love to get help on how to do that?

This was my best guess:

```typescript
declare namespace JSONAPISerializer {
  class JSONAPISerializer {
  }
}
```

As you probably know that doesn't work. 😅

Reply:
No problem, it’s confusing. Let me explain:

> This is how you use the lib: `import JSONAPISerializer from 'json-api-serializer'`

This is not as true as you may think it is. You also use the lib like this:

```js
const JSONAPISerializer = require('json-api-serializer');
```

These two things mean _completely different things_. As a definition author, your job is to describe the exports, not to describe how users will write imports. The JS says `module.exports =`, which means you have to write `export =`. Anything else is wrong, because it will break _one_ of the two imports above. You need them _both_ to work.

> I tried to figure out how to add a namespace and still make that signature work.

The namespace should be a sibling of the class. They merge, and the merged symbol is what gets exported:

```ts
declare namespace JSONAPISerializer {
  export type Foo = {}
}
class JSONAPISerializer {}

export = JSONAPISerializer;
```

Usage:

```ts
// With esModuleInterop (TS or JS):
import JSONAPISerializer from 'json-api-serializer';
new JSONAPISerializer();
let x: JSONAPISerializer.Foo = {}; // (type annotation TS only)

// Without (TS only):
import JSONAPISerializer = require('json-api-serializer');
new JSONAPISerializer();
let x: JSONAPISerializer.Foo = {};

// In JS with CommonJS:
const JSONAPISerializer = require('json-api-serializer');
new JSONAPISerializer();
/** @type {JSONAPISerializer.Foo} */
let x = {};
```

Reply:
Ah, got it! Thanks for the help!

I've updated the types and removed the lint/compiler options I added. :)